exercises/practice/wordy/.meta/example.cairo

```diff
@@ -0,0 +1,203 @@
+    let words: Array<ByteArray> = split_string_into_words(word);
+    if words.len() < 3 {
+        return (0, false);
+    }
+
+    if words[0] != @"What" || words[1] != @"is" {
+        return (0, false);
+    }
+
+    let mut result: i32 = match words.get(2) {
+        Option::Some(s) => {
+            match convert_ByteArray_to_int(s.unbox()) {
+                Option::Some(s) => s,
+                Option::None => { return (0, false); }
+            }
+        },
+        Option::None => { return (0, false); },
+    };
+
+    let mut i = 3;
+    while i < words.len() {
+        let op = parse_operator(words.at(i));
+        i += 1;
+
+        if op == Operator::Multiply || op == Operator::Divide {
+            if i >= words.len() || words[i] != @"by" {
+                result = 0;
+                break;
+            }
+            i += 1;
+        }
+
+        let num: i32 = match words.get(i) {
+            Option::Some(s) => {
+                match convert_ByteArray_to_int(s.unbox()) {
+                    Option::Some(s) => s,
+                    Option::None => {
+                        result = 0;
+                        break;
+                    }
+                }
+            },
+            Option::None => {
+                result = 0;
+                break;
+            },
+        };
+
+        result = match op {
+            Operator::Plus => result + num,
+            Operator::Minus => result - num,
+            Operator::Multiply => result * num,
+            Operator::Divide => {
+                if num == 0 {
+                    result = 0;
+                    break;
+                }
+                result / num
+            },
+            Operator::Invalid => {
+                result = 0;
+                break;
+            },
+        };
+
+        i += 1;
+    };
+
+    if (result == 0) {
+        (result, false)
+    } else {
+        (result, true)
+    }
+}
+
+
+fn split_string_into_words(long_string: ByteArray) -> Array<ByteArray> {
+    let mut words: Array<ByteArray> = ArrayTrait::new();
+    let mut current_word = "";
+
+    let mut i = 0;
+    while i < long_string.len() {
+        let byte = long_string[i];
+        if byte == 32 {
+            if current_word.len() > 0 {
+                words.append(current_word);
+                current_word = "";
+            }
+        } else if (byte == 63) {
+            break;
+        } else {
+            current_word.append_byte(byte);
+        }
+        i += 1;
+    };
+
+    if current_word.len() > 0 {
+        words.append(current_word);
+    }
+
+    words
+}
+
+
+fn convert_ByteArray_to_int(num: @ByteArray) -> Option<i32> {
+    let mut result: Option<i32> = Option::Some(0);
+    let mut size = num.len();
+    let mut i = 0;
+    let mut base = 0;
+
+    let mut isSigned = false;
+    if num.at(i).unwrap() == 45 {
+        isSigned = true;
+        size -= 1;
+    }
+    let num: ByteArray = num.rev();
```

Comment:
The current logic here is:
- take a num string, e.g. `123`
- reverse it, `321`
- init result to `0`
- multiply each digit with 10 to the power of the digit's index
- sum it all together.

But there is a more efficient approach that doesn't require `pow(i)`:
- take a num string, e.g. `123`
- init result to `0`
- for each digit, multiple the result by `10` and add the digit
- get same result as before

Reply:
Thanks for the review. I'd make changes to the efficient one.